x/stakeibc/migrations/v3/convert_test.go

```diff
@@ -0,0 +1,188 @@
+package v3
+
+import (
+	"testing"
+
+	sdkmath "cosmossdk.io/math"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/stretchr/testify/require"
+
+	oldstakeibctypes "github.com/Stride-Labs/stride/v13/x/stakeibc/migrations/v3/types"
+	newstakeibctypes "github.com/Stride-Labs/stride/v13/x/stakeibc/types"
+)
+
+func TestConvertToNewValidator(t *testing.T) {
+	name := "name"
+	address := "address"
+	weight := uint64(3)
+	delegation := sdk.NewInt(4)
+	sharesToTokensRate := sdk.NewDec(5)
+	tvl := sdk.NewInt(1_000_000)
+	slashCheckpoint := sdk.NewInt(10_000) // 1% of TVL
+
+	// First convert a validator with no exchange rate
+	// It should get filled in with the default
+	oldValidator := oldstakeibctypes.Validator{
+		Name:          name,
+		Address:       address,
+		DelegationAmt: delegation,
+		Weight:        weight,
+	}
+	expectedNewValidator := newstakeibctypes.Validator{
+		Name:                        name,
+		Address:                     address,
+		Weight:                      weight,
+		Delegation:                  delegation,
+		SlashQueryProgressTracker:   sdkmath.ZeroInt(),
+		SlashQueryCheckpoint:        slashCheckpoint,
+		DelegationChangesInProgress: 0,
+		SharesToTokensRate:          DefaultExchangeRate,
+		SlashQueryInProgress:        false,
+	}
+
+	actualNewValidator := convertToNewValidator(oldValidator, tvl)
+	require.Equal(t, expectedNewValidator, actualNewValidator)
+
+	// Then add an exchange rate and convert again
+	oldValidator.InternalExchangeRate = &oldstakeibctypes.ValidatorExchangeRate{
+		InternalTokensToSharesRate: sharesToTokensRate,
+		EpochNumber:                1,
+	}
+	expectedNewValidator.SharesToTokensRate = sharesToTokensRate
+
+	actualNewValidator = convertToNewValidator(oldValidator, tvl)
+	require.Equal(t, expectedNewValidator, actualNewValidator)
+}
+
+func TestConvertToNewHostZone(t *testing.T) {
+	chainId := "chain"
+	connectionId := "connection"
+	bechPrefix := "bech"
+	channelId := "channel"
+	ibcDenom := "ibc"
+	hostDenom := "host"
+
+	depositAddress := "address"
+	withdrawalAddress := "withdrawal"
+	feeAddress := "fee"
+	delegationAddress := "delegation"
+	redemptionAddress := "redemption"
+
+	redemptionRate := sdk.NewDec(1)
+	lastRedemptionRate := sdk.NewDec(2)
+	minRedemptionRate := sdk.MustNewDecFromStr("0.95")
+	maxRedemptionRate := sdk.MustNewDecFromStr("1.25")
+	unbondingFrequency := uint64(4)
+	unbondingPeriod := uint64(21)
```

Comment:
I recall seeing unit tests for the `UP=(UF-1 * 7)` conversion formula. If we have those somewhere prob not necessary to add them here. Otherwise might be good to add a few unit tests around that conversion here.

Reply:
we've got a suite of tests on the reverse [here](https://github.com/Stride-Labs/stride/blob/main/x/stakeibc/types/host_zone_test.go). Imo, I think this test case is sufficient since all our mainnet unbonding frequencies are either 3, 4, or 5